docs/GetStarted/getting_started_android_cmake.md

```diff
@@ -0,0 +1,174 @@
+# Getting Started on Android with CMake
+
+<!--
+Notes to those updating this guide:
+
+    * This document should be __simple__ and cover essential items only.
+      Notes for optional components should go in separate files.
+-->
+
+This guide walks through cross-compiling IREE core runtime towards the Android
+platform. Cross-compiling IREE compilers towards Android is not supported
+at the moment.
+
+Cross-compilation involves both a *host* platform a *target* platform. One
+invokes compiler toolchains on the host platform to generate libraries and
+executables that can be run on the target platform.
+
+## Prerequisites
+
+### Set up host development environment
+
+The host platform should have been set up for developing IREE.
+Right now only Linux is supported; Windows and macOS support is coming.
+Please make sure you have followed the steps in
+[Get Started on Linux with CMake](./getting_started_linux_cmake.md).
+
+### Install Android NDK
+
+Android NDK provides compiler toolchains for compiling C/C++ code to target
+Android. You can download it
+[here](https://developer.android.com/ndk/downloads).
+
+Alternatively, if you have installed
+[Android Studio](https://developer.android.com/studio), you can follow [this
+guide](https://developer.android.com/studio/projects/install-ndk) to install
+Android NDK.
+
+After downloading, it is recommended to `export` the `ANDROID_NDK` environment
+variable pointing to the directory in your shell's rc file.
+
+### Install Android Debug Bridge (ADB)
+
+Search your Linux distro's package manager to install `adb`. For example,
+on Ubuntu:
+
+```shell
+$ sudo apt install adb
+```
+
+## Build
+
+Configure:
+
+```shell
+# Assuming in IREE source root
+
+$ cmake -G Ninja -B build-android/  \
+    -DCMAKE_TOOLCHAIN_FILE=$ANDROID_NDK/build/cmake/android.toolchain.cmake \
+    -DANDROID_ABI="arm64-v8a" -DANDROID_PLATFORM=android-29 \
```

Comment:
FYI.

I got stuck here since my SDK/NDK installs are old (I'll go through and update / configure in more detail later)

```
CMake Error at /usr/local/google/home/scotttodd/Android/Sdk/ndk-bundle/build/cmake/android.toolchain.cmake:235 (message):
  android-29 is above the maximum supported version 28.  Choose a supported
  API level or set ANDROID_PLATFORM to "latest".
Call Stack (most recent call first):
  /usr/share/cmake-3.16/Modules/CMakeDetermineSystem.cmake:93 (include)
  CMakeLists.txt:26 (project)
```

changed to `latest` and got this:

```
CMake Error at third_party/llvm-project/llvm/cmake/modules/CheckCompilerVersion.cmake:97 (message):
  libstdc++ version must be at least 5.1
```

Reply:
Yup. Generally it's better to always use the latest Android NDK release. Let me put more words around this in the doc. Thanks for pointing out! :)